src/gtclang/Unittest/ParsingComparison.cpp

```diff
@@ -0,0 +1,244 @@
+  }
+};
+
+struct NestableFunctions : public dawn::codegen::MemberFunction {
+  NestableFunctions(const dawn::Twine& type, const dawn::Twine& name, std::stringstream& s,
+                    int il = 0)
+      : MemberFunction(type, name, s, il) {}
+
+  NestableFunctions addFunction(const dawn::Twine& returntype, const dawn::Twine& name) {
+    NestableFunctions nf(returntype, name, ss(), IndentLevel + 1);
+    return nf;
+  }
+
+  NestableFunctions addVerticalRegion(const dawn::Twine& min, const dawn::Twine& max) {
+    NestableFunctions nf(dawn::Twine::createNull(), "vertical_region", ss(), IndentLevel + 1);
+    nf.addArg(min).addArg(max);
+    nf.startBody();
+    return nf;
+  }
+
+  NestableFunctions addIfStatement(const dawn::Twine& ifCondition) {
+    NestableFunctions ifStatement(dawn::Twine::createNull(), "if", ss(), IndentLevel + 1);
+    ifStatement.addArg(ifCondition);
+    ifStatement.startBody();
+    return ifStatement;
+  }
+};
+
+struct VerticalRegion : public NestableFunctions {
+  VerticalRegion(const dawn::Twine& name, std::stringstream& s, int il = 0)
+      : NestableFunctions(dawn::Twine::createNull(), name, s, il) {}
+};
+
+struct StencilBase : public dawn::codegen::Structure {
+  StencilBase(const dawn::Twine& type, const dawn::Twine& name, std::stringstream& s)
+      : Structure(type.str().c_str(), name, s) {}
+
+  Statement addStorage(const dawn::Twine& memberName) {
+    Statement member(ss(), IndentLevel + 1);
+    member << "storage"
+           << " " << memberName;
+    return member;
+  }
+
+  NestableFunctions addDoMethod(const dawn::Twine& type) {
+    NestableFunctions nf(type, "Do", ss(), IndentLevel + 1);
+    return nf;
+  }
+
+  Statement addOffset(const dawn::Twine& offsetName) {
+    Statement member(ss(), IndentLevel + 1);
+    member << "offset"
+           << " " << offsetName;
+    return member;
+  }
+};
+
+struct StencilFunction : public StencilBase {
+  StencilFunction(const dawn::Twine& name, std::stringstream& s)
+      : StencilBase("stencil_function", name, s) {}
+
+  virtual NestableFunctions addDoMethod(const dawn::Twine& type) {
+    NestableFunctions nf(type, "Do", ss(), IndentLevel + 1);
+    nf.startBody();
+    return nf;
+  }
+
+  virtual ~StencilFunction() {}
+};
+
+struct Stencil : public StencilBase {
+  Stencil(const dawn::Twine& name, std::stringstream& s) : StencilBase("stencil", name, s) {}
+};
+
+struct Globals : public StencilBase {
+  Globals(std::stringstream& s) : StencilBase("globals", dawn::Twine::createNull(), s) {}
+  virtual NestableFunctions addDoMethod(const dawn::Twine& type) = delete;
+  virtual Statement addOffset(const dawn::Twine& offsetName) = delete;
+  virtual ~Globals() {}
+};
+
+class FileWriter {
+public:
+  FileWriter(std::string testPath, std::string filename) {
+    filename_ = UnittestEnvironment::getSingleton().getFileManager().dataPath() +
+                "/gtclang/Frontend/" + testPath;
+    llvm::sys::fs::create_directories(filename_);
```

Comment:
This functionality belongs to the `FileManager`, it should make sure the generated path is valid (or throw an exception otherwise).